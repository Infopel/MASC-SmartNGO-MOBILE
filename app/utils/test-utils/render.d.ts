export type RenderOptions = {
  wrapper?: React.ComponentType<any>
  createNodeMock?: (element: React.ReactElement) => any
}
export type AllTheProvidersProps = { children: React.ReactNode }
