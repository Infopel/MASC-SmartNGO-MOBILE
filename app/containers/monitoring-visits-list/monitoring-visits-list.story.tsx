import * as React from "react"
import { storiesOf } from "@storybook/react-native"
import { StoryScreen, Story, UseCase } from "../../../storybook/views"
import { color } from "../../theme"
import { MonitoringVisitsList } from "./monitoring-visits-list.component"

storiesOf("MonitoringVisitsList", module)
  .addDecorator((fn) => <StoryScreen>{fn()}</StoryScreen>)
  .add("Style Presets", () => (
    <Story>
      <UseCase text="Primary" usage="The primary.">
        <MonitoringVisitsList style={{ backgroundColor: color.error }} />
      </UseCase>
    </Story>
  ))
