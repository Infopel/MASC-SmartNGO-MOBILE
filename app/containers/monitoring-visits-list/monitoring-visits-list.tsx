import { Q } from "@nozbe/watermelondb"
import Database from "@nozbe/watermelondb/Database"
import { withDatabase } from "@nozbe/watermelondb/DatabaseProvider"
import withObservables from "@nozbe/with-observables"
import { BaseList } from "components/base-list/base-list"
import { MonitoringVisitsListItem } from "components/monitoring-visits-list-item/monitoring-visits-list-item"
import * as React from "react"
import { MonitoringVisit } from "storage/models/monitoring-visit"

export interface MonitoringVisitsListProps {
  initiativeId: string
  items?: MonitoringVisit[]
}

/**
 * Describe your component here
 */
 function Container({ items }: MonitoringVisitsListProps) {


  return (
    <BaseList data={items} renderItem={({ item }) => <MonitoringVisitsListItem item={item} />} />
  )
}

export const MonitoringVisitsListContainer = withDatabase(withObservables(['initiativeId','database'], ({ initiativeId, database }: { database: Database; initiativeId: string }) => ({ items: database.get<MonitoringVisit>(MonitoringVisit.table).query(Q.where("idIniciativa", initiativeId)) }))(Container))
