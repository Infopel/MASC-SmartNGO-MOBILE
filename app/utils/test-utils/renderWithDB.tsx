jest.mock("storage/adapter.ts")
import DatabaseProvider from "@nozbe/watermelondb/DatabaseProvider"
import { render } from "@testing-library/react-native"
import { default as React } from "react"
import { database } from "storage/database"
import { NativeBaseProvider as ThemeProvider } from "theme"
import { AllTheProvidersProps, RenderOptions } from "./render"

function AllTheProvidersWithDB({ children }: AllTheProvidersProps) {
    return (
        <DatabaseProvider database={database}>
            <ThemeProvider>{children}</ThemeProvider>
        </DatabaseProvider>
    )
}

export function renderWithDB<T extends any>(ui: React.ReactElement<T>, options?: RenderOptions) {
    return render(ui, { wrapper: AllTheProvidersWithDB, ...options })
}


export class DiagnosticError extends Error {
    name: string = "Diagnostic error"
  }
  