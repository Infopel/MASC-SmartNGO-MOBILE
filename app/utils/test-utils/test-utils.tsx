import { render } from "@testing-library/react-native"
import React from "react"
import { NativeBaseProvider as ThemeProvider } from "theme"
import { AllTheProvidersProps, RenderOptions } from "./render"

function AllTheProviders({ children }: AllTheProvidersProps) {
  return (
    <ThemeProvider>{children}</ThemeProvider>
  )
}

function customRender<T extends any>(ui: React.ReactElement<T>, options?: RenderOptions) {
  return render(ui, { wrapper: AllTheProviders, ...options })
}

// re-export everything
export * from "@testing-library/react-native"
// override render method
export { customRender as render }

