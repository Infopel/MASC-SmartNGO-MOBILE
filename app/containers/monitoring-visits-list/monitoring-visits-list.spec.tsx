import { fireEvent, render } from "utils/test-utils/test-utils"
import * as React from "react"
import { MonitoringVisitsListContainer } from "./monitoring-visits-list"

describe("<MonitoringVisitsListContainer>",()=>{
 it("render", async function () {
    const {toJSON} = render(<MonitoringVisitsListContainer></MonitoringVisitsListContainer>)
    
    expect(toJSON()).toMatchSnapshot()
  })
})
  