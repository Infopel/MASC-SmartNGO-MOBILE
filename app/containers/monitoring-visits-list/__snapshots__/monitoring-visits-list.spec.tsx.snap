// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MonitoringVisitsListContainer> render 1`] = `
<View
  style={
    Object {
      "padding": 16,
    }
  }
  testID="monitoringVisitsListContainer"
>
  <Text
    dataSet={Object {}}
    style={
      Array [
        Object {
          "backgroundColor": undefined,
          "color": "#171717",
          "fontFamily": undefined,
          "fontSize": 14,
          "fontStyle": "normal",
          "fontWeight": "400",
          "letterSpacing": 0,
          "lineHeight": 21,
          "textDecorationLine": undefined,
        },
        Object {
          "color": "#FBA928",
          "fontFamily": "Helvetica",
          "fontSize": 14,
        },
      ]
    }
    testID="monitoringVisitsListContainerTestId"
  >
    Hello
  </Text>
</View>
`;
